pyteal/compiler/optimizer/optimizer.py

```diff
@@ -0,0 +1,72 @@
+from dataclasses import dataclass
+from typing import List, cast
+
+from pyteal.ir.tealcomponent import TealComponent
+from pyteal.ir.tealop import TealOp
+from pyteal.ir.ops import Op
+
+
+@dataclass
+class OptimizeOptions:
+    """An object which specifies the optimizations to be performed.
+    Args:
+
+        use_slot_to_stack (optional): cancel contiguous store/load operations
+            that have no load dependencies elsewhere.
+        iterate_optimizations (optional): repeat the optimizations until the
+            code is no longer changed."""
+
+    use_slot_to_stack: bool = False
+    iterate_optimizations: bool = False
+
+
+def _has_load_dependencies(teal: List[TealComponent], slot: int, pos: int):
+    for i, op in enumerate(teal):
+        if i == pos:
+            continue
+
+        if type(op) == TealOp and op.op == Op.load and op.args[0] == slot:
+            return True
+
+    return False
+
+
+def _apply_slot_to_stack(teal: List[TealComponent]) -> List[TealComponent]:
+    remove = [False] * len(teal)
+    for i, op in enumerate(teal[:-1]):
+        if type(op) != TealOp or op.op != Op.store:
+            continue
+
+        next_op = teal[i + 1]
+        if type(next_op) != TealOp or next_op.op != Op.load:
+            continue
+
+        if op.args[0] != next_op.args[0]:
+            continue
+
+        if not _has_load_dependencies(teal, cast(int, op.args[0]), i + 1):
+            remove[i] = remove[i + 1] = True
+
+    return [op for i, op in enumerate(teal) if not remove[i]]
```

Comment:
nit: it would be more space-efficient to make `remove` a set of indexes rather than a list that's always as long as the program.